src/handlers/process-subscriptions.ts

```diff
@@ -0,0 +1,115 @@
+  if (isNil(walletData)) {
+    const sponsorLogOptions = buildLogOptions('additional', { sponsor: sponsorAddress }, providerLogOptions);
+    utils.logger.warn('Failed to fetch sponsor wallet or transaction count', sponsorLogOptions);
+    return;
+  }
+
+  const { sponsorWallet, transactionCount } = walletData;
+
+  const sponsorWalletLogOptions = buildLogOptions(
+    'additional',
+    { sponsorWallet: shortenAddress(sponsorWallet.address) },
+    providerLogOptions
+  );
+  utils.logger.logPending(transactionCountLogs, sponsorWalletLogOptions);
+  utils.logger.info(`Processing ${subscriptions.length} subscription(s)`, sponsorWalletLogOptions);
+
+  const processSponsorWalletResult = await processSponsorWallet(
+    airnodeWallet,
+    contracts['DapiServer'],
+    gasTarget,
+    subscriptions,
+    sponsorWallet,
+    voidSigner,
+    transactionCount
+  );
+
+  processSponsorWalletResult.forEach(([logs, data]) => {
+    const subscriptionLogOptions = buildLogOptions('additional', { subscriptionId: data.id }, sponsorWalletLogOptions);
+    utils.logger.logPending(logs, subscriptionLogOptions);
+  });
+};
+
+export const handler = async ({
+  providerSponsorSubscription,
+  baseLogOptions,
+}: {
+  providerSponsorSubscription: ProviderSponsorSubscriptions;
+  baseLogOptions: utils.LogOptions;
+}): Promise<AWSHandlerResponse> => {
+  const airnodeConfig = goSync(loadAirnodeConfig);
+  if (!airnodeConfig.success) {
+    utils.logger.error(airnodeConfig.error.message);
+    throw airnodeConfig.error;
+  }
+
+  const airnodeWallet = ethers.Wallet.fromMnemonic(airnodeConfig.data.nodeSettings.airnodeWalletMnemonic);
+
+  // Initialize provider specific data
+  const [logs, evmProviderState] = await initializeProvider(
+    providerSponsorSubscription.providerGroup.chainConfig,
+    providerSponsorSubscription.providerGroup.providerUrl || ''
+  );
+  const providerLogOptions = buildLogOptions(
+    'meta',
+    {
+      chainId: providerSponsorSubscription.providerGroup.chainId,
+      providerName: providerSponsorSubscription.providerGroup.providerName,
+    },
+    baseLogOptions
+  );
+  utils.logger.logPending(logs, providerLogOptions);
+  if (isNil(evmProviderState)) {
+    const message = 'Failed to initialize provider';
+    utils.logger.warn(message, providerLogOptions);
+
+    return {
+      statusCode: 500,
+      ok: false,
+      message: `Failed to initialize provider: ${providerSponsorSubscription.providerGroup.providerName} for chain: ${providerSponsorSubscription.providerGroup.chainId}`,
+    };
+  }
+
+  await processSubscriptions(
+    {
+      ...providerSponsorSubscription,
+      providerState: { ...providerSponsorSubscription.providerGroup, airnodeWallet, ...evmProviderState },
+    },
+    baseLogOptions
+  );
+
+  return {
+    statusCode: 200,
+    ok: true,
+    message: `Processing subscriptions for sponsorAddress: ${providerSponsorSubscription.sponsorAddress} has finished`,
```

Comment:
I've moved config loading and  `initializeProvider` inside this function.

Reply:
Just being curious, would that mean that we will be loading the same config file for each provider in the config?

Reply:
Yes we are loading it for each provider to get the `airnodeWalletMnemonic`. I figured it's probably not a good idea to send it in the `lambda.invoke` **Payload** :smile: